src/Illuminate/Auth/Authenticatable.php

```diff
@@ -25,6 +25,18 @@ public function getAuthPassword()
     }
 
     /**
+     * Automatically encrypt password if not done yet.
+     *
+     * @param $pass
+     */
+    public function setPasswordAttribute($value)
+    {
+        $value = app('hash')->needsRehash($value) ? bcrypt($value) : $value;
```

Comment:
Well, if the value being set **is** already a hash, it shouldn't be encrypted once more.. So this line checks that and encrypts if not already encrypted.